pkg/crc/machine/kubeconfig.go

```diff
@@ -242,33 +242,42 @@ func contains(arr []string, str string) bool {
 	return false
 }
 
-func mergeKubeConfigFile(kubeConfigFile string) error {
-	globalConfigPath, kubeConf1, err := getGlobalKubeConfig()
-	if err != nil {
-		return err
+func mergeKubeConfigFile(kubeConfigFile, globalConfigFile string) error {
+	var kubeConf1 *api.Config
+	var err error
+	if globalConfigFile != "" {
+		kubeConf1, err = clientcmd.LoadFromFile(globalConfigFile)
+		if err != nil {
+			return err
+		}
+	} else {
+		_, kubeConf1, err = getGlobalKubeConfig()
+		if err != nil {
+			return err
+		}
 	}
 	kubeConf2, err := clientcmd.LoadFromFile(kubeConfigFile)
 	if err != nil {
 		return err
 	}
 	// Merge the kubeConf2 to globalConfig
-	mergedConfig, err := clientcmd.NewDefaultClientConfig(*kubeConf1, &clientcmd.ConfigOverrides{}).ConfigAccess().GetStartingConfig()
-	if err != nil {
-		return err
-	}
 
 	for name, cluster := range kubeConf2.Clusters {
-		mergedConfig.Clusters[name] = cluster
+		kubeConf1.Clusters[name] = cluster
 	}
 
 	for name, authInfo := range kubeConf2.AuthInfos {
-		mergedConfig.AuthInfos[name] = authInfo
+		kubeConf1.AuthInfos[name] = authInfo
 	}
 
 	for name, context := range kubeConf2.Contexts {
-		mergedConfig.Contexts[name] = context
+		kubeConf1.Contexts[name] = context
 	}
```

Comment:
This function ends with `return clientcmd.ModifyConfig(configAccess, *kubeConf1, true)` which is documented as:
```
// ModifyConfig takes a Config object, iterates through Clusters, AuthInfos, and Contexts, uses the LocationOfOrigin if specified or uses the default destination file to write the results into.
```

I was hoping that using `ModifyConfig` would allow us to remove these iterations over `Clusters`, `AuthInfos` and `Contexts`?

Reply:
I couldn't find a way to do it without these iterations